zlib-rs/src/inflate.rs

```diff
@@ -14,15 +15,34 @@ use crate::{
     DEF_WBITS, MAX_WBITS, MIN_WBITS,
 };
 
+use crate::crc32;
+
 use self::{
     bitreader::BitReader,
     inftrees::{inflate_table, CodeType, InflateTable},
     window::Window,
 };
 
-// TODO should only be used by tests; only export when running tests
-pub const INFLATE_STATE_SIZE: usize = core::mem::size_of::<crate::inflate::State>();
+/// TODO: Move to separate file?
+#[repr(C)]
+#[derive(Copy, Clone, Debug)]
+pub struct GzipHeader {
+    pub text: libc::c_int,
+    pub time: crate::c_api::z_size,
+    pub xflags: libc::c_int,
+    pub os: libc::c_int,
+    pub extra: *mut crate::c_api::Bytef,
+    pub extra_len: crate::c_api::uInt,
+    pub extra_max: crate::c_api::uInt,
+    pub name: *mut crate::c_api::Bytef,
+    pub name_max: crate::c_api::uInt,
+    pub comment: *mut crate::c_api::Bytef,
+    pub comment_max: crate::c_api::uInt,
+    pub hcrc: libc::c_int,
+    pub done: libc::c_int,
+}
```

Comment:
this struct does not add any additional safety or guarantees, so I think we can just use `gz_header` everywhere. Just saves code.